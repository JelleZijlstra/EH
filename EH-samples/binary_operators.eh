#!/usr/bin/ehi
class Foo {
  operator_and = func: rhs {
    echo 'Binary and with ' + rhs
  }
  operator_or = func: rhs {
    echo 'Binary or with ' + rhs
  }
  operator_xor = func: rhs {
    echo 'Binary xor with ' + rhs
  }
  operator_tilde = func: {
    echo 'Binary negation'
  }
  operator_uminus = func: {
    echo 'Unary minus'
  }
  operator_bang = func: {
    echo 'Negation'
  }
}
f = Foo.new:
$f & 2
$f ^ 3
$f | 4
~f
-f
!f
