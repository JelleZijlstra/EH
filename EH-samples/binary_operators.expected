Binary and with 2
Binary xor with 3
Binary or with 4
Binary negation
Unary minus
Negation
